wade-review.md

```diff
@@ -0,0 +1,596 @@
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            streams.close(Duration.ofSeconds(5));
+            latch.countDown();
+        }));
+}
+
+Then we’ll set up a countdown latch so it waits for other threads before it starts…
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+        final KafkaStreams streams = new KafkaStreams(gitHubPrRatio.topology(props), props);
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            streams.close(Duration.ofSeconds(5));
+            latch.countDown();
+        }));
+
+        try {
+            streams.start();
+            latch.await();
+        } 
+    }
+}
+
+Then, in our try block, we’ll actually start the stream.
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+        final KafkaStreams streams = new KafkaStreams(gitHubPrRatio.topology(props), props);
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            streams.close(Duration.ofSeconds(5));
+            latch.countDown();
+        }));
+
+        try {
+            streams.start();
+            latch.await();
+        } catch (final Throwable e) {
+            e.printStackTrace();
+            System.exit(1);
+        }
+        System.exit(0);
+    }
+}
+
+Our catch block will print the error and the stack trace. 
+
+
+## The topology
+
+```java
+    public Topology topology(Properties properties) {
+
+}
+```
+
+Ok, so,  what’s a topology? It’s  a definition of the computational logic we need to process our data stream. What does it look like? Let’s make one.
+
+
+```java
+    public Topology topology(Properties properties) {
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        builder.stream(INPUT_TOPIC, Consumed.with(Serdes.String(), jsonNodeSerde))
+    }
+
+
+}
+```
+
+First, I’ll instantiate the builder, then I’ll create the stream. We specify the input topic in the Consumed.with method, as well as the deserializer we’ll use for the key and value.
```

Comment:
Seems not quite correct. The input topic isn't specified "in" the `Consumed.with` method, it's specified before it as part of the `builder.stream` params.

Reply:
https://github.com/Cerchie/notes-for-gh-video/pull/2/commits/19087d9f12c2e5eaca0fb53964cda832f92180ab